### ISSUE TITLE
Docking ports

### ISSUE THREAD

**Issue by Tarkov** [zmaster587/AdvancedRocketry#271]
Not sure if I am doing something wrong.

I build the space station, go there and place a docking port with the cross looking outside, then I write a name ID, "colombo" for example. 

In the overworld I build a module and put the docking port with the cross looking outside as well, and write "colombo" as target. I build it in the station assembler like the space station using a satellite bay and the space station ID chip that I already have. 

Put it in a rocket like the space station and send it. The rocket goes to orbit and does not come. When I travel to the station, it is as I left it before, no signs of the second module, and the docking port is still in its place.

It is a new test world with just Advanced Rocketry and Buildcraft in 1.7.10

**Comment by zmaster587** [comment]
Do you have a seat or a guidance computer on the rocket?

**Comment by Tarkov** [comment]
No, just tanks, engines and satellite bay.

**Comment by zmaster587** [comment]
investigating

**Comment by Tarkov** [comment]
By the way, when writing the name in the docking port, the key functions are still available, if you press a,w,s,d,e or what ever, it moves or opens the inventory, or the function display.

**Comment by zmaster587** [comment]
Fixed the key issue with e5934ee919db86c141cc16bd222b9bcae75a196b

**Comment by zmaster587** [comment]
Somehow the directions got reveresed so the modules spawn in facing the wrong way.  But I've not been able to get it to not spawn in at all.

**Comment by zmaster587** [comment]
if you break and replace then name the docking port on the station does it work?

**Comment by Tarkov** [comment]
I think all depends on how you face the docking ports in both dimensions. If you face the port looking south in the space station, you need to build the new module facing north in the overworld to fit it, otherwise it does not spawn or do it incorrectly. If the new module finds blocks in its position in the station it does not spawn.

**Comment by zmaster587** [comment]
Yeah, I found that the algorithm that calculated rotation was reversed for
some reason, if the module built on the ground faced the same way or the
opposite direction with respect to the module in orbit it would spawn
correctly, otherwise unless the ground module faced east/west it would be
reversed.  I'm used to programs having Z face up and checked against the
wrong direction.  But I could have sworn I tested this case before  release
and had everything working properly.  That commit should fix this though.
But even if the rotation is wrong at the very least the station pad should
be deleted.
